packages/react/src/runtime/provider/provider-browser.tsx

```diff
@@ -25,9 +25,28 @@ if (process.env.NODE_ENV === 'development' && isNodeEnvironment()) {
 const inserted: Record<string, true> = {};
 
 /**
- * Noops on the client
+ * Iterates through all found style elements generated when server side rendering.
+ *
+ * @param cb
+ */
+const forEachSSRdStyleElement = (cb: (element: HTMLStyleElement) => void) => {
+  const ssrStyles = document.querySelectorAll<HTMLStyleElement>('style[data-cmpld]');
+  for (let i = 0; i < ssrStyles.length; i++) {
+    cb(ssrStyles[i]);
+  }
+};
+
+// Move all found server-side rendered style elements to the head before React hydration happens.
+forEachSSRdStyleElement((styleElement) => {
```

Comment:
Thanks for moving this here ❤️. This gives clear picture that this runs before React actually kicks in.